File: lesson_3/src/main/java/com/skifer/transport/Transport.java
package com.skifer.transport;

/**
 * Абстрактный класс траснпорта
 */
abstract public class Transport {

    /**
     * Цвет транспорта
     */
    protected String color;

    /**
     * Название транспорта
     */
    protected String name;

    /**
     * Колличество колёс
     */
    protected Integer wheels;

    /**
     * Возможная вместительность
     */
    protected Integer capacity;

    /**
     * Абстрактный класс траснпорта
     * @param color цвет
     * @param name название
     * @param wheels колличество колёс
     * @param capacity вместимость
     */
    public Transport(String color, String name, Integer wheels, Integer capacity) {
        this.color = color;
        this.name = name;
        this.wheels = wheels;
        this.capacity = capacity;
    }

    public abstract void getInfo();

    @Override
    public String toString() {
        return "{color='" + color + '\'' +
                ", name='" + name + '\'' +
                ", wheels=" + wheels +
                ", capacity=" + capacity +
                '}';
    }
}

File: lesson_5/src/main/kotlin/com/skifer/transport/Transport.kt
package com.skifer.transport

/**
 * Simple Transport abstract class
 */
sealed class Transport {

    /**
     * Model name
     */
    abstract val model: String

    /**
     * Transport color
     */
    abstract val color: Color

    /**
     * Vehicle Identification Number
     */
    abstract val VIN: String

    /**
     * Simple Car class
     */
    data class Car (
        override val model: String,
        override val color: Color,
        override val VIN: String,
         /** Is this car dirty? */
        var needsCleaning: Boolean = false
        ): Transport() {
        override fun toString(): String {
            return "Car(model='$model', color=$color, VIN='$VIN', needsCleaning=$needsCleaning)"
        }
    }

    /**
     * Simple Bike class
     */
    data class MotorCycle(
        override val model: String,
        override val color: Color,
        override val VIN: String,
        /** Does this motorcycle need refueling? */
        var isOutOfFuel: Boolean = false
        ): Transport() {
        override fun toString(): String {
            return "MotorCycle(model='$model', color=$color, VIN='$VIN', isOutOfFuel=$isOutOfFuel)"
        }
    }

}
